enhancements/cluster-logging/cluster-logging-log-forwarding.md

```diff
@@ -104,198 +156,160 @@ This is often required for industries such as the US public sector, healthcare o
-       ca_file /var/run/ocp-collector/secrets/elasticsearch/cacert
-       <buffer>
-		@type file
-		path '/var/lib/fluentd/elasticsearch'
-       </buffer>
-     </store>
-  </match>
-  <match **>
-     @type copy
-     <store>
-       @type elasticsearch
-       @id elasticsearch
-       client_key /var/run/ocp-collector/secrets/elasticsearch/key
-       client_cert /var/run/ocp-collector/secrets/elasticsearch/cert
-       ca_file /var/run/ocp-collector/secrets/elasticsearch/cacert
-       retry_tag "retry_elasticsearch"
-     </store>
- </match>
-</label>
-<label @SECUREFORWARD_OFFCLUSTER>
-   ...
-</label>
+apiVersion: "logging.openshift.io/v1alpha1"
+kind: "Forwarder"
+metadata:
+    namespace: my-tenant-namespace # Logs are restricted to this namespace.
+spec:
+   outputs:
+   - name: secure-tool
+     url: fluent+tls://mytool.cluster.local:
+     secret:
+        name: my_secrets # Must contain keys tls.key, tls.cert and ca.cert 
+
+  pipelines:
+   - outputs: [ secure-tool ]
 ```
 
-Following is a diagram which identifies the workflow of a message through fluentd as captured by the above configuration.
+#### As a cluster administrator, I want to clearly separate where the logging stack forwards infrastructure and/or audit related logs.
+
+```
+apiVersion: "logging.openshift.io/v1alpha1"
+kind: "Forwarder"
+metadata:
+  namespace: cluster-logging # Global forwarder
+  pipelines:
+   - selector: { source: [Infrastructure] }
+     outputs: [ elasticsearch+https://infra.cluster.local ]
+   - selector: { source: [Application] }
+     outputs: [ elasticsearch+https://app.cluster.local ]
+   - selector: { source: [Audit] }
+     outputs: [ syslog+tls://mysyslogservice.example.net ]
+```
 
-![https://docs.google.com/drawings/d/18FD4c1pBz4e8eJIH37yUwM40ew69g8h1cqPEfOLsuyY/edit?usp=sharing](images/log-forwarding-message-flow.png)
+#### As a Red Hat SRE who operates OSD clusters, I want a mechanism to protect my configuration (e.g. audit log forwarding, infra logs) from non SRE administrators  of OSD but at the same time give them the opportunity to configure their own log forwarding for applications.
+
+Create global forwarder with RBAC control restricted to SRE admins. Users are free to create their own forwarders (global or namespace-restricted) independent of the SRE forwarder.
+
+```
+apiVersion: "logging.openshift.io/v1alpha1"
+kind: "Forwarder"
+metadata:
+  namespace: cluster-logging # Global forwarder
+spec:
+   - selector: { source: [Audit, Infrastructure] }
+     outputs: [ elasticsearch+https://sre.destination.example.com ]
+```
+
+#### I want to configure log forwarding to include/exclude logs on k8s labels.
+
+This applies equally to global or namespace-restricted forwarders.
+
+```
+apiVersion: "logging.openshift.io/v1alpha1"
+kind: "Forwarder"
+spec:
+  outputs: ... define outputs called 'everything' and 'critical'
+  pipelines:
+   - outputs: [ everything ] # All logs.
+   - selector: { labels: { importance: critical } }
```

Comment:
convention elsewhere is that 'selector' is a map that are labels on pods

Reply:
Pushed past GA, but for the future: There are 2 kinds of selectors, maps and expressions (strings) Expressions are more powerful, so I'd be tempted to do only that, but maps are older and more widespread. I've yet to find any canonical statement about what's the "right" way to do it, so I opted for both: selector: { labels: {name:value ...}, expressions: [exprs...]} but I'm not sure that's the "right" thing either.